.github/workflows/build-test-publish-images.yml

```diff
@@ -0,0 +1,254 @@
+      PYTHON_VER: ${{ matrix.PYTHON_VER }}
+      RAPIDS_VER: ${{ needs.compute-matrix.outputs.RAPIDS_VER }}
+      DASK_SQL_VER: ${{ matrix.DASK_SQL_VER }}
+      BASE_TAG:
+        "rapidsai/${{ needs.compute-matrix.outputs.BASE_IMAGE_REPO }}:\
+        ${{ needs.compute-matrix.outputs.BASE_TAG_PREFIX }}\
+        ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
+        ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
+        cuda${{ matrix.CUDA_VER }}-\
+        ${{ matrix.LINUX_VER }}-\
+        py${{ matrix.PYTHON_VER }}"
+      NOTEBOOKS_TAG:
+        "rapidsai/${{ needs.compute-matrix.outputs.NOTEBOOKS_IMAGE_REPO }}:\
+        ${{ needs.compute-matrix.outputs.NOTEBOOKS_TAG_PREFIX }}\
+        ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
+        ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
+        cuda${{ matrix.CUDA_VER }}-\
+        ${{ matrix.LINUX_VER }}-\
+        py${{ matrix.PYTHON_VER }}"
+  test:
+    needs: [build, compute-matrix]
+    # TODO: nightly tests
+    if: inputs.build_type == 'pull-request'
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.TEST_MATRIX) }}
+      fail-fast: false
+    secrets: inherit
+    uses: ./.github/workflows/test-image.yml
+    with:
+      BUILD_TYPE: ${{ inputs.build_type }}
+      ARCH: ${{ matrix.ARCH }}
+      GPU: ${{ matrix.GPU }}
+      DRIVER: ${{ matrix.DRIVER }}
+      NOTEBOOKS_TAG:
+        "rapidsai/${{ needs.compute-matrix.outputs.NOTEBOOKS_IMAGE_REPO }}:\
+        ${{ needs.compute-matrix.outputs.NOTEBOOKS_TAG_PREFIX }}\
+        ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
+        ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
+        cuda${{ matrix.CUDA_VER }}-\
+        ${{ matrix.LINUX_VER }}-\
+        py${{ matrix.PYTHON_VER }}-\
+        ${{ matrix.ARCH }}"
+  build-multiarch-manifest:
+    if: inputs.build_type == 'branch'
+    needs: [build, test, compute-matrix]
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.MATRIX) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.GPUCIBOT_DOCKERHUB_USER }}
+          password: ${{ secrets.GPUCIBOT_DOCKERHUB_TOKEN }}
+      - name: Create multiarch manifest
+        run: |
+          base_source_tags=()
+          notebooks_source_tags=()
+
+          base_tag_array=(
+            rapidsai/${{ needs.compute-matrix.outputs.BASE_IMAGE_REPO }}:
+            ${{ needs.compute-matrix.outputs.BASE_TAG_PREFIX }}
+            ${{ needs.compute-matrix.outputs.RAPIDS_VER }}
+            ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-
+            cuda${{ matrix.CUDA_VER }}-
+            ${{ matrix.LINUX_VER }}-
+            py${{ matrix.PYTHON_VER }}
+          )
+          base_tag="$(printf %s "${base_tag_array[@]}" $'\n')" # Converts array to string w/o spaces
```

Comment:
I made this change based on the comment here: https://github.com/rapidsai/docker/pull/545#discussion_r1264190712

It's just an effort to make it more readable. I would like to find a better way to construct these strings though since they are (mostly) reused several times.